pyerrors/covobs.py

```diff
@@ -44,13 +33,43 @@ def __init__(self, mean, cov, name, pos=None, grad=None):
             else:
                 if pos > self.N:
                     raise Exception('pos %d too large for covariance matrix with dimension %dx%d!' % (pos, self.N, self.N))
-            self.grad = np.zeros((self.N, 1))
-            self.grad[pos] = 1.
+            self._grad = np.zeros((self.N, 1))
+            self._grad[pos] = 1.
         else:
-            self.grad = np.array(grad)
+            self.set_grad(grad)
         self.value = mean
 
     def errsq(self):
         """ Return the variance (= square of the error) of the Covobs
         """
         return float(np.dot(np.transpose(self.grad), np.dot(self.cov, self.grad)))
+
+    def set_cov(self, cov):
```

Comment:
I would vote for hidding the `set` routines

Reply:
Agreed. One could even think about using a double underscore for the variables, to make (accidental) changes even more difficult. What do you think?

Reply:
Double underscores could (possibly) cause problems within classes (see [PEP8](https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables)). The main reason why I advocate for leading underscores in functions which are irrelevant for users is that the auto generated documentation also omits them. For the instance attributes like `_value` my idea was to inform the user when attempting to overwrite `value`.